service/dataset.py

```diff
@@ -48,23 +42,65 @@ def read(dataset_request):
     except:
         # TODO: Temp fix for testing - Do better cleanup if secret missing
         # dataset["dataset_type"]["token"] = {'name':dataset_name,'value':42}
-        pass 
-    
+        pass
+
+    return dataset
+
+def readreleased(dataset_request):
```

Comment:
We should add a github issue, or preliminary support her, for clients to create a fake user id guid. This will allow us to better test read released for users before we actually have the authorization code path working. You can pass the guid in as a header for now.

https://stackoverflow.com/questions/29386995/how-to-get-http-headers-in-flask can be used to check for the header. 

This should help us get a jumpstart on how to handle an actual Authorization mechanism.

For example:
generate ids 0-9 for each of the separate clients.
register dataset takes an optional list of valid ids 5-9
read requests fail for 1-4
read released costs budget the first time but not subsequent calls for 0-9

Reply:
nice system test added with this scenario in tests/dataset/test_smoke.py